beacon-chain/blockchain/kzg/kzg.go

```diff
@@ -0,0 +1,154 @@
+	if err != nil {
+		return [48]byte{}, err
+	}
+	return Proof(proof), nil
+}
+
+func ComputeCellsAndKZGProofs(blob *Blob) ([ckzg4844.CellsPerExtBlob]Cell, [ckzg4844.CellsPerExtBlob]Proof, error) {
+	ckzgBlob := ckzg4844.Blob(*blob)
+	_cells, _proofs, err := ckzg4844.ComputeCellsAndKZGProofs(&ckzgBlob)
+	if err != nil {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, [ckzg4844.CellsPerExtBlob]Proof{}, err
+	}
+
+	// Convert Cells and Proofs to types defined in this package
+	var cells [ckzg4844.CellsPerExtBlob]Cell
+	for i := range _cells {
+		cells[i] = Cell(_cells[i])
+	}
+
+	var proofs [ckzg4844.CellsPerExtBlob]Proof
+	for i, proof := range _proofs {
+		proofs[i] = Proof(proof)
+	}
+
+	return cells, proofs, nil
+}
+
+func VerifyCellKZGProof(commitmentBytes Bytes48, cellId uint64, cell Cell, proofBytes Bytes48) (bool, error) {
+	return ckzg4844.VerifyCellKZGProof(commitmentBytes, cellId, ckzg4844.Cell(cell), proofBytes)
+}
+
+func VerifyCellKZGProofBatch(commitmentsBytes []Bytes48, rowIndices, columnIndices []uint64, _cells []Cell, proofsBytes []Bytes48) (bool, error) {
+	// Convert `Cell` type to `ckzg4844.Cell`
+	ckzgCells := make([]ckzg4844.Cell, len(_cells))
+	for i := range _cells {
+		ckzgCells[i] = ckzg4844.Cell(_cells[i])
+	}
+
+	return ckzg4844.VerifyCellKZGProofBatch(commitmentsBytes, rowIndices, columnIndices, ckzgCells, proofsBytes)
+}
+
+func RecoverAllCells(cellIds []uint64, _cells []Cell) ([ckzg4844.CellsPerExtBlob]Cell, error) {
+	// Convert `Cell` type to `ckzg4844.Cell`
+	ckzgCells := make([]ckzg4844.Cell, len(_cells))
+	for i := range _cells {
+		ckzgCells[i] = ckzg4844.Cell(_cells[i])
+	}
+
+	recoveredCells, err := ckzg4844.RecoverAllCells(cellIds, ckzgCells)
+	if err != nil {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, err
+	}
+
+	// This should never happen, we return an error instead of panicking.
+	if len(recoveredCells) != ckzg4844.CellsPerExtBlob {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, errors.New("recovered cells length is not equal to CellsPerExtBlob")
+	}
+
+	// Convert `ckzg4844.Cell` type to `Cell`
+	var ret [ckzg4844.CellsPerExtBlob]Cell
+	for i := range recoveredCells {
+		ret[i] = Cell(recoveredCells[i])
+	}
+	return ret, nil
+}
+
+// Recovers the cells and compute the KZG Proofs associated with the cells.
+//
+// This method will supercede the `RecoverAllCells` and `CellsToBlob` methods.
+func RecoverCellsAndKZGProofs(cellIds []uint64, _cells []Cell) ([ckzg4844.CellsPerExtBlob]Cell, [ckzg4844.CellsPerExtBlob]Proof, error) {
+	// First recover all of the cells
+	recoveredCells, err := RecoverAllCells(cellIds, _cells)
+	if err != nil {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, [ckzg4844.CellsPerExtBlob]Proof{}, err
+	}
+
+	// Extract the Blob from all of the Cells
+	blob, err := CellsToBlob(recoveredCells)
+	if err != nil {
+		return [ckzg4844.CellsPerExtBlob]Cell{}, [ckzg4844.CellsPerExtBlob]Proof{}, err
+	}
+
+	// Compute all of the cells and KZG proofs
+	return ComputeCellsAndKZGProofs(&blob)
+}
```

Comment:
This method is not being used -- the most recent commit of the crypto libraries will no longe expose RecoverAllCells and CellsToBlob, so I've added this method tentatively, so that we can modify the codebase preemptively in anticipation for that breaking change